src/app/core/entity-components/entity-list/list-paginator/list-paginator.component.ts

```diff
@@ -0,0 +1,137 @@
+import { Component, ViewChild, Input, SimpleChanges } from "@angular/core";
+import { Entity } from "../../../entity/entity";
+import { MatPaginator, PageEvent } from "@angular/material/paginator";
+import { MatTableDataSource } from "@angular/material/table";
+import { User } from "app/core/user/user";
+import { SessionService } from "app/core/session/session-service/session.service";
+import { EntityMapperService } from "app/core/entity/entity-mapper.service";
+import { getUrlWithoutParams } from "app/utils/utils";
+import { Router } from "@angular/router";
+
+@Component({
+  selector: "app-list-paginator",
+  templateUrl: "./list-paginator.component.html",
+  styleUrls: ["./list-paginator.component.scss"],
+})
+export class ListPaginatorComponent<E extends Entity> {
+  @Input() dataSource: MatTableDataSource<E>;
+
+  @ViewChild(MatPaginator) paginator: MatPaginator;
+
+
+  user: User;
+  paginatorPageSize: number = 10;
+  paginatorPageSizeBeforeToggle: number = 10;
+  paginatorPageIndex: number = 0;
+  showAllToggle: boolean = false;
+
+  // This key is used to save the pagination settings on the user entity
+  readonly paginatorKey: string;
+
+  constructor(
+    private sessionService: SessionService,
+    private entityMapperService: EntityMapperService,
+    private router: Router,
+  ) {
+    this.paginatorKey = getUrlWithoutParams(this.router);
+  }
+
+  ngOnInit() {
+    console.log("PaginatorKey: " + this.paginatorKey);
+    this.user = this.sessionService.getCurrentUser();
+    // Use URl as key to save pagination settings
+    this.paginatorPageSize =
+      this.user.paginatorSettingsPageSize[this.paginatorKey] ||
+      this.paginatorPageSize;
+    this.paginatorPageIndex =
+      this.user.paginatorSettingsPageIndex[this.paginatorKey] ||
+      this.paginatorPageIndex;
+    console.log("OnInit aufgerufen. Length ist " + this.dataSource.data.length);
+  }
+
+  ngOnChanges(changes: SimpleChanges): void {
+    this.showAllToggle =
+      this.paginatorPageSize >= this.dataSource.data.length;
+      console.log("OnChanges aufgerufen. Length ist " + this.dataSource.data.length);
+  }
+
+  ngAfterViewInit() {
+    this.dataSource.paginator = this.paginator;
+    setTimeout(() => {
+      this.paginator.pageIndex = this.paginatorPageIndex;
+      this.paginator.page.next({
+        pageIndex: this.paginator.pageIndex,
+        pageSize: this.paginator.pageSize,
+        length: this.paginator.length,
+      });
+      console.log("AfterViewInit mit Timeout aufgerufen. Length ist " + this.dataSource.data.length);
+    });
+    console.log("AfterViewInit aufgerufen. Length ist " + this.dataSource.data.length);
+  }
+
+  onPaginateChange(event: PageEvent) {
+    this.paginatorPageSize = event.pageSize;
+    this.paginatorPageIndex = event.pageIndex;
+    this.updateUserPaginationSettings();
+    this.showAllToggle =
```

Comment:
Why does this have to be recalculated when the page is changed? Is this also triggered when the page size is increased?

Reply:
This is also triggered when the page size is changed or increased. I think that calling the update method is necessary because the changes might need to be saved to the user.